src/analysis/special_functions/log.lean

```diff
@@ -172,6 +172,34 @@ begin
   { rintro (rfl|rfl|rfl); simp only [log_one, log_zero, log_neg_eq_log], }
 end
 
+lemma log_le_sub_one_of_nonneg (x : ℝ) (hx : 0 < x) : log x ≤ x - 1 :=
+begin
+  have h := add_one_le_exp (log x),
+  rw exp_log hx at h,
+  rw le_sub_iff_add_le,
+  exact h,
+end
+
+lemma log_div_self_decreasing (x y : ℝ) (hex : exp 1 ≤ x) (hxy : x ≤ y) :
+  log y / y ≤ log x / x :=
+begin
+  have x_pos : 0 < x := lt_of_lt_of_le (exp_pos 1) hex,
```

Comment:
minor golf: `have x_pos : 0 < x := (exp_pos 1).trans_le hex`, and similar below.
And for `x_ne_zero` you can say `x_pos.ne'`.

Reply:
thanks!